client/my-sites/plans/jetpack-plans/product-store/index.tsx

```diff
@@ -10,13 +15,40 @@ import './style.scss';
 const ProductStore: React.FC< ProductStoreProps > = ( {
 	enableUserLicensesDialog,
 	urlQueryArgs,
+	duration,
 } ) => {
 	const siteId = useSelector( getSelectedSiteId );
+	const currentPlan = useSelector( ( state ) => getSitePlan( state, siteId ) );
+	const currentPlanSlug = currentPlan?.product_slug || null;
+	const { availableProducts, purchasedProducts, includedInPlanProducts } =
+		useGetPlansGridProducts( siteId );
+
+	const allItems = useMemo(
+		() => [
+			...getProductsToDisplay( {
+				duration,
+				availableProducts,
+				purchasedProducts,
+				includedInPlanProducts,
+			} ),
+			...getPlansToDisplay( { duration, currentPlanSlug } ),
+		],
+		[ duration, availableProducts, purchasedProducts, includedInPlanProducts, currentPlanSlug ]
+	);
 
 	return (
 		<div className="jetpack-product-store">
 			{ enableUserLicensesDialog && <UserLicensesDialog siteId={ siteId } /> }
+
+			<div className="jetpack-product-store__pricing-banner">
+				<IntroPricingBanner
+					productSlugs={ allItems.map( ( { productSlug } ) => productSlug ) }
+					siteId={ siteId ?? 'none' }
+				/>
+			</div>
+
```

Comment:
Can we extract this to a separate component like `IntroBanner`/`intro-banner.tsx`?

Reply:
That's good thinking James. I was wondering if it would still be better to extract that logic to a custom hook instead. Something like `useProductSlugs()`

Reply:
The signature of the hook could be something like this:

```ts
function useProductSlugs( { duration, siteId }: SomeType ) {
    //
}
```

Reply:
That is actually a good idea.. 👍 I'll work on the custom hook..

Reply:
@manzoorwanijk I am using this signature and just followed how the rest of the custom hooks are done in the current code. 

```
const useProductSlugs = ( siteId: number | null, duration: Duration ): string[] => {
```

Reply:
Such parameters are not ideal because they can be hard to read at call-sites.
`useProductSlugs( 12345, 'year' )` is not very semantic.

`useProductSlugs( { siteId: 12345, duration: 'year' } )` on the other hand is more readable.

Reply:
I added `ProductSlugsProps` type to make the parameter readable  as suggested. 👍